Top-level declarations in .d.ts files must start with either a 'declare' or 'export' modifier

### Describe the bug

Hello! First of all, thanks for the great library 👍 It's very cool!

But now I'm trying to update wonka from 6.2.5 to 6.2.6 or 6.3.0 on one of my projects and typescript gives me the following error:

```
node_modules/wonka/dist/wonka.d.ts:9:1 - error TS1046: Top-level declarations in .d.ts files must start with either a 'declare' or 'export' modifier.

9 enum TalkbackKind {
  ~~~~


Found 1 error in node_modules/wonka/dist/wonka.d.ts:9
```

I tried to build it locally, but I installed deps using yarn instead of pnpm, so effectively all the latest, and it outputs `declare` for enums correctly. But on published version on npm there's no `declare` keywords before enums. So I assume it may be a problem with some of the dependencies.

I created a repo with simple reproduction, just in case. It has wonka@6.3.0 and gives an error. But if you change it to wonka@6.2.5, there will be no errors.

### Reproduction

https://github.com/faergeek/wonka-tsc-issue-reproduction

### Package version

wonka@6.2.6

### Validations

- [X] I can confirm that this is a bug report, and not a feature request, RFC, question, or discussion, for which GitHub Discussions should be used
- [X] Follow our [Code of Conduct](https://github.com/0no-co/graphql.web/blob/main/CODE_OF_CONDUCT.md)

--- a/.changeset/light-cougars-enjoy.md
+++ b/.changeset/light-cougars-enjoy.md
@@ -1,0 +1,5 @@
+---
+'wonka': patch
+---
+
+Fix missing `declare` keyword on internal ambient enums.

--- a/src/types.d.ts
+++ b/src/types.d.ts
@@ -6,7 +6,7 @@
  * the {@link Start} signal, to tell a {@link Source} to either send a new value (pulling) or stop
  * sending values altogether (cancellation).
  */
-export enum TalkbackKind {
+export declare enum TalkbackKind {
   /** Instructs the {@link Source} to send the next value. */
   Pull = 0,
   /** Instructs the {@link Source} to stop sending values and cancels it. */
@@ -41,7 +41,7 @@ export type TeardownFn = () => void;
  * @see {@link Start} for the data structure of the start signal.
  * @see {@link Push} for the data structure of the push signal, carrying values.
  */
-export enum SignalKind {
+export declare enum SignalKind {
   /**
    * Informs the {@link Sink} that it's being called by a {@link Source}.
    *

